File: algorithm.java/src/main/java/leetcode/p0001/S1.java
package leetcode.p0001;


import java.util.HashMap;

public class S1 implements Solution {
    public int[] twoSum(int[] nums, int target)
    {
        int[] ret = {-1, -1};
        Integer j = -1;
        HashMap<Integer, Integer> map = new HashMap<>(nums.length);
        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], i);
        }
        for (int i = 0; i < nums.length; i++) {
            if ((j = map.get(target - nums[i])) != null && j.intValue() != i) {
                ret[0] = i;
                ret[1] = j;
                break;
            }
        }
        return ret;
    }

}

File: algorithm.python/p0001/s1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        l = len(nums)
        for i in range(0, l):
            v = target -  nums[i]
            if d.has_key(v):
                return [d[v], i]
            else:
                d[nums[i]] = i

